src/win/fs.c

```diff
@@ -2065,24 +2153,59 @@ INLINE static DWORD fs__stat_directory(WCHAR* path, uv_stat_t* statbuf,
   stat_info.CreationTime.QuadPart = dir_info.CreationTime.QuadPart;
   stat_info.LastAccessTime.QuadPart = dir_info.LastAccessTime.QuadPart;
   stat_info.LastWriteTime.QuadPart = dir_info.LastWriteTime.QuadPart;
-  if (stat_info.FileAttributes & FILE_ATTRIBUTE_REPARSE_POINT) {
```

Comment:
There should not need to be any changes to `fs__stat_directory` function, as you're introducing a TOCTOU bug by doing so